File: Scripts/pd_glider.py
from glife.base import *

rule()

def collision (i, j):
	return pentadecathlon + glider[i + 11] (-8 + j, -10, flip)

all = pattern ()

for i in range (-7, 8):
	for j in range (-9, 10):
		all += collision (i, j) (100 * i, 100 * j)

### all.save ("pd_glider.lif", "Exploring 'pentadecathlon + glider' collisions.")

all.display ("pd_glider")

File: Scripts/Lua/pd-glider.lua
-- Creates a large set of pentadecathlon + glider collisions.
-- Based on pd_glider.py from PLife (http://plife.sourceforge.net/).

local g = golly()
local gp = require "gplus"
local gpo = require "gplus.objects"

g.new("pd-glider")
-- best to create empty universe before setting rule
-- to avoid converting an existing pattern (slow if large)
g.setrule("B3/S23")

local function collision(i, j)
    return gpo.pentadecathlon + gpo.glider[i + 11].t(-8 + j, -10, gp.flip)
end

local all = gp.pattern()
for i = -7, 7 do
    for j = -9, 9 do
        all = all + collision(i, j).t(100 * i, 100 * j)
    end
end

all.display("") -- don't change name
